Cython/Distutils/build_ext.py

```diff
@@ -10,34 +11,158 @@
 else:
     from distutils.command.build_ext import build_ext as _build_ext
 
+try:
+    from __builtin__ import basestring
+except ImportError:
+    basestring = str
 
-class new_build_ext(_build_ext, object):
 
+class build_ext(_build_ext, object):
+
+    sep_by = _build_ext.sep_by
     user_options = _build_ext.user_options[:]
     boolean_options = _build_ext.boolean_options[:]
 
     user_options.extend([
+        ('cython-cplus', None,
+         "generate C++ source files"),
+        ('cython-create-listing', None,
+         "write errors to a listing file"),
+        ('cython-line-directives', None,
+         "emit source line directives"),
+        ('cython-include-dirs=', None,
+         "path to the Cython include files" + sep_by),
         ('cython-c-in-temp', None,
-             "put generated C files in temp directory"),
+         "put generated C files in temp directory"),
+        ('cython-gen-pxi', None,
+            "generate .pxi file for public declarations"),
+        ('cython-directives=', None,
+            "compiler directive overrides"),
+        ('cython-gdb', None,
+         "generate debug information for cygdb"),
+        ('cython-compile-time-env', None,
+            "cython compile time environment"),
+
+        # For backwards compatibility.
```

Comment:
A question: Can we remove pyrex-* options in cython 3.0 (only from new_build_ext, they will be still present in old_build_ext)? Does it make sense?

Reply:
Yes, I think that's the right time to do that.

Reply:
Removed.